packages/o-spreadsheet-engine/src/plugins/ui_stateful/sheetview.ts

```diff
@@ -522,6 +538,19 @@ export class SheetViewPlugin extends UIPlugin {
     return { ...rect, x: rect.x + this.gridOffsetX, y: rect.y + this.gridOffsetY };
   }
 
+  /**
+   * Computes the coordinates and size to draw the zone on the canvas after it has been zoomed
+   */
+  getVisibleRectWithZoom(zone: Zone): Rect {
+    const zoom = this.getViewportZoomLevel();
+    const rect = this.getVisibleRectWithoutHeaders(zone);
+    rect.width = rect.width * zoom;
+    rect.height = rect.height * zoom;
+    rect.x = rect.x * zoom + this.gridOffsetX + HEADER_WIDTH * (zoom - 1);
+    rect.y = rect.y * zoom + this.gridOffsetY + HEADER_HEIGHT * (zoom - 1);
```

Comment:
we'll probably want to support zoom level for dashboards as well, especially in mobile